### PR TITLE
[AIRFLOW-2916] Arg `verify` for AwsHook() & S3 sensors/operators

### PR DESCRIPTION
**Pull request by XD-DENG**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-2916
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### `Background`
The `AwsHook()` and `S3`-related operators/sensors are depending on package `boto3`.

In `boto3`, when we initiate a `client` or a `resource`, argument `verify` is provided (https://boto3.readthedocs.io/en/latest/reference/core/session.html ).

### `Why do we need this`
It is useful when
- users want to use a different CA cert bundle than the one used by `botocore`.
- users want to have **'--no-verify-ssl'** ([link](https://docs.aws.amazon.com/cli/latest/reference/)), otherwise error `[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed` may appear. **This is especially useful when we're using on-premises S3 or other implementations of object storage, like IBM's Cloud Object Storage**.

### `Backward Compatibility`
The default value here is always `None`, which is also the default value in `boto3`, so that backward compatibility is ensured too.

### `Others`
Another SSL related argument in `boto3`, `use_ssl`, can be ignored here because it will be ignored by `boto3` if `endpoint_url` is provided (`endpoint_url` is always provided in `AwsHook()`).

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by Fokko — CHANGES_REQUESTED** [review]
I'm not a big fan of the ability to set them, because I think you should always use proper SSL certificates. Shouldn't be set to `True` by default?

**Comment by XD-DENG** [comment]
Hi @Fokko . I do agree that we should always have proper SSL certificates. But this feature may be quite necessary when
- if we want to use a different CA cert bundle than the one used by `botocore` (argument `verify` can be either boolean or string. When it's string, it can be path/to/cert/bundle.pem).
- for my own case, my infra team didn't set proper SSL for the on-prem UAT S3. Before they set it up for me, I have to use `--no-verify-ssl`, i.e. put `verify` as `False`. I believe I'm not the only one who have this sort of issue.

Regarding default value: In `boto3`, default value of `verify` is `None`, and actually it's equivalent to `True` here (please refer to [boto3 session reference](https://boto3.readthedocs.io/en/latest/reference/core/session.html)). when we run 
```
session.client(client_type, endpoint_url=endpoint_url)
```
it's equivalent to
```
session.client(client_type, endpoint_url=endpoint_url, verify=None)
``` 
and SSL certificates would be verified.
To keep consistent with `boto3`, I also put `None` as default value here.

Hope this clarifies.

**Comment by nsreenivas** [comment]
+1

I believe standard practice for most SSL communication is to provide a way to turn it off but enable by default. S3 API and LDAP protocol support communication with and without SSL. Organisations who want to disable insecure connection will generally block the port that allows insecure communication (LDAP 389 port). Additionally, we have development environments where users want to test tools without having to deal with things like TLS and Kerberos.

The other point is when we are in an on-premise environment using tools like Minio or IBM Cloud Object Store, we need to be able to pass certs that aren't the default AWS certificates. This is a requirement for running Airflow securely in an on-premise cloud.

I would suggest defaulting to SSL enabled but allow the opportunity for users to disable it if required. As the AWS CLI does, an warning could be thrown telling users that noSSL is an insecure option.

**Review by Fokko — APPROVED** [review]

**Comment by XD-DENG** [comment]
Hi @Fokko , a gentle ping. May this PR be merged? Thanks!

**Comment by XD-DENG** [comment]
Hi @Fokko , friendly ping: JIRA ticket of this PR is not closed yet. Understand that committers need to set up the fix version for it when closing in JIRA, so I leave it to you to close it?

https://issues.apache.org/jira/browse/AIRFLOW-2916

Thanks.

**Comment by Fokko** [comment]
Yes, thanks you @XD-DENG

**Pull request merged**
